File: FizzBuzz.js
function fizzBuzz(words = ["Fizz", "Buzz"], length = 100) {
    for (var i = 1; i <= length; i++) {
        var output = "";
        for (var j = 0; j < words.length; j++) {
            if (i % ((j + 1) * 2 + 1) === 0) {
                output += words[j]
            }
        }
        if (output === "") {
            output = i;
        }
        console.log(output);
    }
}

fizzBuzz(["Fizz", "Buzz", "Bazz"], 1000)

File: FizzBuzz.lua
function fizzbuzz(words, length)
    words = words or {"Fizz", "Buzz"}
    length = length or 100
    for i = 1, length do
        output = ""
        for j, value in pairs(words) do
            if (i % ((j) * 2 + 1)) == 0 then
                output = output .. value
            end
        end
        if output == "" then
            output = i
        end
        print(output)
    end
end

fizzbuzz({"Fizz", "Buzz", "Bazz"}, 1000)
